statefun-sdk-java/src/main/java/org/apache/flink/statefun/sdk/java/testing/Envelope.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.statefun.sdk.java.testing;
+
+import java.time.Duration;
+import java.util.Objects;
+import org.apache.flink.statefun.sdk.java.message.Message;
+
+/**
+ * A utility class that wraps a {@link Message} and the {@link Duration} after which it was sent by
+ * a {@link org.apache.flink.statefun.sdk.java.StatefulFunction}. It is used by the {@link
+ * TestContext}.
+ */
+public class Envelope {
```

Comment:
I like the idea of an envelope, but I think that this class is a bit confusing as it, requires a delay, and generally a bit tricky to extend in the future (make it capture a `context` method).
How about we'll just split it to `SentEgressMessage` , `SentMessage`, and `SentDelayedMessage` ?

They don't have to implement a common interface) but just 3 separate classes to capture the result of 3 different `context` invocations.

Reply:
Done.